src/stub/core_impl/base.rs

```diff
@@ -34,6 +34,43 @@ impl<T: Target, C: Connection> GdbStubImpl<T, C> {
         Ok(tid)
     }
 
+    pub(crate) fn get_sane_current_pid(
+        &mut self,
+        target: &mut T,
+    ) -> Result<Pid, Error<T::Error, C::Error>> {
+        match target.base_ops() {
+            BaseOps::SingleThread(_) => Ok(FAKE_PID),
+            BaseOps::MultiThread(ops) => ops.active_pid().map_err(Error::TargetError),
+        }
+    }
+
+    // res.write_str("S00")?;
+    // HandlerStatus::Handled
+
+    #[inline]
```

Comment:
```suggestion
    #[inline(always)]
```

Empirically, I've found that bare `#[inline]` call aren't super useful - it's only with the addition of `always` or `never` that the compiler actually respects your requests. Otherwise, it's just a suggestion (that often gets ignored)

Reply:
though, while you're here... can you double-check that my intuition here is correct?

i.e: is the code smaller if you force the inline, or does it actually make the binary larger?

Reply:
There's no difference between the two builds:
```sh
user@Ondo:/opt/Xous/gdbstub$ git diff
diff --git a/src/stub/core_impl/base.rs b/src/stub/core_impl/base.rs
index c392289..1cf42ee 100644
--- a/src/stub/core_impl/base.rs
+++ b/src/stub/core_impl/base.rs
@@ -44,7 +44,7 @@ impl<T: Target, C: Connection> GdbStubImpl<T, C> {
         }
     }

-    #[inline(always)]
+    #[inline]
     pub(crate) fn report_reasonable_stop_reason(
         &mut self,
         res: &mut ResponseWriter<'_, C>,
user@Ondo:/opt/Xous/gdbstub$ ./example_no_std/check_size.sh > inline.txt
   Compiling gdbstub v0.6.4 (/opt/Xous/gdbstub)
   Compiling gdbstub_arch v0.2.4 (/opt/Xous/gdbstub/gdbstub_arch)
   Compiling gdbstub-nostd v0.1.0 (/opt/Xous/gdbstub/example_no_std)
    Finished release [optimized + debuginfo] target(s) in 3.17s
    Finished release [optimized + debuginfo] target(s) in 0.00s
    Analyzing target/release/gdbstub-nostd

user@Ondo:/opt/Xous/gdbstub$ git checkout src/stub/core_impl/base.rs
Updated 1 path from the index
user@Ondo:/opt/Xous/gdbstub$ ./example_no_std/check_size.sh > inline-always.txt
   Compiling gdbstub v0.6.4 (/opt/Xous/gdbstub)
   Compiling gdbstub_arch v0.2.4 (/opt/Xous/gdbstub/gdbstub_arch)
   Compiling gdbstub-nostd v0.1.0 (/opt/Xous/gdbstub/example_no_std)
    Finished release [optimized + debuginfo] target(s) in 3.09s
    Finished release [optimized + debuginfo] target(s) in 0.00s
    Analyzing target/release/gdbstub-nostd

user@Ondo:/opt/Xous/gdbstub$ diff -u inline-always.txt inline.txt
user@Ondo:/opt/Xous/gdbstub$
```